proto/xudp2p.proto

```diff
@@ -0,0 +1,116 @@
+syntax = "proto3";
+
+import "annotations.proto";
+
+package xudp2p;
+
+message Header {
+    string id = 1 [json_name = "id"];
+    string reqId = 2 [json_name = "req_id"];
```

Comment:
I noticed that in the `validate` methods, we are checking to make sure that `reqId` and `hash` are undefined for packets that don't use them. It seems like it would be better to only use those fields on the message definitions for packets that actually use them, therefore we don't need to explicitly validate in our code. So instead of having a `Header` with all three fields like we did in the prior JSON implementation, we just put these fields directly on messages that actually use them. For example, `GetNodesPacket` would have `id` as its only field. But `NodesPacket` would have `id`, `reqId`, and `hash` along with the actual nodes.

Reply:
Sounds fine, just the it won't give us a clear distinction between the body and the header. Also, when we'll want to hash the body, we need to make sure we don't get confused about which fields we want to put in. 

If it sounds ok to you, i'll do the change.

Reply:
Those are good points. Another possibility would be defining different Header messages such as `Header` & `ResponseHeader` and using those where appropriate. But I'm thinking this way might be good after all, so go with whatever you think is best.